lib/authorizer/ntlm.js

```diff
@@ -150,7 +149,7 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done -
      */
     pre: function (auth, done) {
-        !auth.get(STATE) && auth.set(STATE, STATES.INITIALIZED);
+        !auth.getAuthState(NTLM) && auth.setAuthState(NTLM, STATES.INITIALIZED);
```

Comment:
All auths should be using the state now instead of `auth.set`. Can we ensure that there's no use of `auth.set` for updating the state?

Reply:
@coditva This auth object is specific to the auth state only, i.e. if a req is using multiple req for auth and wants to share the state b/w the requests, that time this will be used.
that's the reason not we can't set anything other than it
other auth's don't need an auth state